assignments/prototypes.js

```diff
@@ -16,13 +16,40 @@
   * destroy() // prototype method that returns: `${this.name} was removed from the game.`
 */
 
+function GameObject(object) {
+  this.createdAt = object.createdAt;
+  this.name = object.name;
+  this.dimensions = object.dimensions;
+}
+
+GameObject.prototype.destroy = function() {
+  return `${this.name} was removed from the game.`
+}
+
 /*
   === CharacterStats ===
   * healthPoints
   * takeDamage() // prototype method -> returns the string '<object name> took damage.'
   * should inherit destroy() from GameObject's prototype
 */
 
+function CharacterStats(newStats) {
+  GameObject.call(this, newStats);
+  this.healthPoints = newStats.healthPoints;
+}
+
+/* Why did I have to add the line for CharacterStats to inherit from GameObject before
+creating the takeDamage() method in the prototype? When I had the order reversed and I viewed
+the function in the console, the takeDamage() method was not a part of the function. But when I organized
+the lines like I have below, it is there. */
```

Comment:
when you inherit from GameObject, it is basically inheriting all the object methods that are in the object prototype as well as the methods you have added i.e. `destroy`. if you inherit  them after you add a method it will overwrite the method you have created, so, inheriting before you create the method you inherit all the methods as well as being able to add one